### ISSUE TITLE
Empty lanes not fixed or linter

### ISSUE THREAD

**Issue by lokofoko** [sqlfluff/sqlfluff#5899]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```
select distinct
    p.clip
    , rpa.itemdataid
    , p.composite_property_linkage_key
    , l.fips
    , l.apn
    , l.sequence
    , rpa.firstname
    , rpa.lastname
    , rpa.mobile

from mlg_snowflake.external_data.rasani_phone_append as rpa

inner join mlg_snowflake.data_integrations.lead as l on rpa.referencenr = l.refnr

left outer join mlg_snowflake.data_integrations.property_basic_v2 as p on l.fips = p.fips_code and l.apn = p.unformatted_apn and l.sequence = p.apn_sequence_number

where mobile != ''

    and not exists (select * from data_integrations.owner_detail as od where od.corelogicdataid = rpa.itemdataid and od.first_name = rpa.firstname and od.last_name = rpa.lastname)
;
```

For some reason sqlfluff doesn't want to remove these empty lines between from and join and where and between select columns and from.

### Expected Behaviour

It should eliminate empty lines between code lines if it within one statement.

### Observed Behaviour

Bad formatting.

### How to reproduce

You have the code and config. Just run it together

### Dialect

snowflake

### Version

SQLFluff v3.0.6

### Configuration

[sqlfluff]
dialect = snowflake
templater = dbt
exclude_rules = layout.long_lines, ambiguous.column_count, structure.column_order
processes = 2
runaway_limit = 10
max_line_length = 80
indent_unit = space

[sqlfluff:indentation]
tab_space_size = 4
indented_joins = False
indented_using_on = True
indented_on_contents = False
allow_implicit_indents = True

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = leading
spacing_after = single

[sqlfluff:rules:convention.coalesce]

[sqlfluff:rules:convention.not_equal]
preferred_not_equal_style = c_style

[sqlfluff:rules:convention.terminator]
multiline_newline = True

[sqlfluff:rules:references.special_chars]
additional_allowed_characters = "" "

[sqlfluff:rules:ambiguous.column_references]
group_by_and_order_by_style = explicit

### aliasing rules
[sqlfluff:rules:aliasing.table]
aliasing = explicit

[sqlfluff:rules:aliasing.column]
aliasing = explicit

[sqlfluff:rules:aliasing.expression]
allow_scalar = True

### capitalisation rules
[sqlfluff:rules:capitalisation.keywords] 
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @lokofoko - thanks for opening this issue.

Sqlfluff is treating that as valid layout, so fix has nothing to change, i.e it is happy with this shape:

```sql
FROM some_table

INNER JOIN other_table ...
```

It sees the JOIN as its own clause line, and the single blank line between clauses is still within the allowed empty-line limit.

You can force `sqlfluff` to remove those new line by adding this to your config:
```ini
[sqlfluff:rules:layout.newlines]
maximum_empty_lines_inside_statements = 0
```

Let me know if you have more questions

**Comment by peterbud** [comment]
Closing due to lack of response
